plugins/modules/ios_service.py

```diff
@@ -0,0 +1,425 @@
+                type: bool
+            type: dict
+        type: list
+      udp_small_servers:
+        description:
+          - TCP and UDP small servers are servers (daemons, in Unix parlance) that run in the
+            router which are useful for diagnostics.
+        suboptions:
+          enable:
+            description: Enable small UDP servers (e.g., ECHO)
+            type: bool
+          max_servers:
+            description:
+              - Set number of allowable TCP small servers
+              - 1 to 2147483647 or no-limit
+            type: str
+        type: dict
+      unsupported_transceiver:
+        description: enable support for third-party transceivers
+        type: bool
+    type: dict
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOS device by
+        executing the command B(show running-config | section ^service|^no service).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - parsed
+      - gathered
+      - rendered
+    default: merged
+    description:
+      - The state the configuration should be left in.
+      - Refer to examples for more details.
+      - The states I(replaced) and I(overridden) have identical
+        behaviour for this module.
+    type: str
+"""
+EXAMPLES = """
+# Using state: merged
+
+# Before state:
+# -------------
+
+# router-ios#show running-config | section ^service|^no service
+# no service timestamps debug uptime
+# no service timestamps log uptime
+
+# Merged play:
+# ------------
+
+- name: Apply the provided configuration
+  cisco.ios.ios_service:
+    config:
+      tcp_keepalives_in: true
+      tcp_keepalives_out: true
+      timestamps:
+        - msg: debug
+          enable: true
+          timestamp: datetime
+        - msg: log
+          enable: true
+          timestamp: datetime
+      pad: false
+      password_encryption: true
+    state: merged
+
+# Commands Fired:
+# ---------------
+
+# "commands": [
+#         "no service pad",
+#         "service password-encryption",
+#         "service tcp-keepalives-in",
+#         "service tcp-keepalives-out",
+#         "service timestamps debug datetime",
+#         "service timestamps log datetime"
+# ]
+
+# After state:
+# ------------
+
+# switch-ios#sh run | sec ^service|^no service
+# no service pad
+# service tcp-keepalives-in
+# service tcp-keepalives-out
+# service timestamps debug datetime
+# service timestamps log datetime
+# service password-encryption
+
+# Using state: deleted
+
+# Before state:
+# -------------
+
+# switch-ios#sh run | sec ^service|^no service
```

Comment:
the before state should be strict as per the command used in facts to generate the configuration for the module.

Reply:
OK: I have updated the comment (and the command use to retrieve the fact for the module is `show running-config all | section ^service ` to be sure to be able to remove some service started by default on some IOS version (service that is recommended to stop by security)